packages/bpk-component-infinite-scroll/src/DataSource.js

```diff
@@ -70,7 +70,9 @@ export class ArrayDataSource<T = any> extends DataSource<T> {
       if (n <= 0) {
         return resolve([]);
       }
-      return resolve(elements.slice(0, index + nElements));
+      const start = Math.max(index, 0);
+      const end = Math.max(start, start + nElements);
+      return resolve(elements.slice(start, end));
```

Comment:
we can may be remove the upper part given that slice will return until the last element of the array even if we ask for more:
`[1].slice(0, 10)` => [1]
`[1].slice(10,10)` => []

Reply:
How do you mean? I don't think `slice(10, 10)` will ever happen here, since we use `start + nElements` so if someone calls `fetchItems(10, 1)` it will be `slice(10, 11)`. And `fetchItems(10, 0)` doesn't really make sense. It should return empty in this case.

Reply:
Just to clarify, the `max` check is just to "protect" to code against someone providing a negative number to `nElements`.

Reply:
sorry, I didn't explain myself, by upper part I meant from line 68 to 72

Reply:
Oh you mean removing the if? I think you're right, on the other hand it's probably clearer/faster to do keep it there instead of "slicing" it when we don't really need to.

Reply:
yes, that's what I meant